pkg/generators/servers.go

```diff
@@ -809,3 +839,22 @@ func (g *ServersGenerator) mapMethodNameToHTTPMethod(name *names.Name) string {
 		return http.MethodGet
 	}
 }
+
+func (g *ServersGenerator) readerName(typ *concepts.Type) string {
+	// see helpers.go file where the following methods are defined.
+	version := typ.Owner()
+	switch {
+	case typ == version.Integer():
+		return "ParseInteger"
+	case typ == version.Float():
+		return "ParseFloat"
+	case typ == version.String():
+		return "ParseString"
+	case typ == version.Date():
+		return "ParseDate"
+	case typ == version.Boolean():
+		return "ParseBoolean"
+	default:
+		return fmt.Sprintf("We do not know how to handle type %v", typ.Name().String())
```

Comment:
Use `g.reporter.Errorf(...)`. That will report the error nicely to the user and will cause the execution of the tool to eventually fail.